### ISSUE TITLE
Batch size

### ISSUE THREAD

**Issue by tGhattas** [goombalab/phi-mamba#2]
Hey! I'm trying to reproduce your experiments, and in the paper its said that the batch size is 2^{15, 16,..} 
a safe assumption to make its not the actual batch size and its couples with gradient accumulation, so something like 
16 (BS)  *  (GradiAccum) 2048 = 2^15, can you confirm? and didn't that mean very few optimizer steps?

Thanks

**Comment by kevinli573** [comment]
Yes, the batch size we mention is the total number of tokens see per optimizer update. For our experiments, we usually had a couple thousand optimizer steps per stage as bs and tokens processed varies from stage to stage.

**Comment by tGhattas** [comment]
@kevinli573 I'm a bit confused regarding "Stages 1 and 2 were trained to 200M steps each while Stage 3 extended to 1B steps" is this the case for the ultimate model phi mamba model? so you performed couple thousand epochs for example for stage 1?

**Comment by kevinli573** [comment]
"Stages 1 and 2 were trained to 200M steps each while Stage 3 extended to 1B steps" in Appendix A was referring to the budget allocated to the hyperparameter sweep for determining learning rate and batch size. The final model was trained with 80M, 160M, and 2.76B tokens for Stage 1, 2, and 3 respectively.

And yes, each stage ended up being in the thousands of epochs.
